packages/agreement-process/test/createAgreement.test.ts

```diff
@@ -658,20 +737,124 @@ describe("create agreement", () => {
     await addOneEService(eservice);
 
     const authData = getRandomAuthData(consumer.id);
-    const apiAgreementPayload: agreementApi.AgreementPayload = {
-      eserviceId: eservice.id,
-      descriptorId: eservice.descriptors[0].id,
-    };
 
     await expect(
-      agreementService.createAgreement(apiAgreementPayload, {
-        authData,
-        correlationId: generateId(),
-        serviceName: "",
-        logger: genericLogger,
-      })
+      agreementService.createAgreement(
+        {
+          eserviceId: eservice.id,
+          descriptorId: eservice.descriptors[0].id,
+        },
+        {
+          authData,
+          correlationId: generateId(),
+          serviceName: "",
+          logger: genericLogger,
+        }
+      )
     ).rejects.toThrowError(
       missingCertifiedAttributesError(descriptor.id, consumer.id)
     );
   });
+  it("should throw missingDelegationId error when there is an active delegation, the requester is the delegate or the delegator, but no delegation id is provided", async () => {
```

Comment:
See the comment I left in the service about this scenario, If that's correct then this should become as follows

```suggestion
  it("should throw cannotCreateAgreementForDelegatedEservice error when there is an active delegation and the requester is the delegator", async () => {
```